src/plugins/vis_type_timeseries/public/application/components/aggs/top_hit.js

```diff
@@ -91,16 +92,20 @@ const getOrderOptions = () => [
   },
 ];
 
+const AGG_WITH_KEY = 'agg_with';
 const ORDER_DATE_RESTRICT_FIELDS = [KBN_FIELD_TYPES.DATE];
 
 const TopHitAggUi = (props) => {
   const { fields, series, panel } = props;
-  const defaults = {
-    size: 1,
-    agg_with: 'noop',
-    order: 'desc',
-  };
-  const model = { ...defaults, ...props.model };
+  const defaults = useMemo(
```

Comment:
in this case it looks like additional optimization. In fact, this will lead to a slight degradation in performance, since each time the `React` caching mechanism will be used to get the value.

If you want you can move it to one level up and create `getDefaults()` method or move it directly in line `108` with removing defaults at all